beastify/popup/choose_beast.js

```diff
@@ -12,33 +36,57 @@ function beastNameToURL(beastName) {
   }
 }
 
-
 /*
 Listen for clicks in the popup.
 
-If the click is not on one of the beasts, return early.
+If the click is on one of the beast buttons:
+  The text content of the node is the name of the beast we want.
 
-Otherwise, the text content of the node is the name of the beast we want.
+  Inject the "beastify.js" content script in the active tab.
 
-Inject the "beastify.js" content script in the active tab.
+  Then get the active tab and send "beastify.js" a message
+  containing the URL to the chosen beast's image.
 
-Then get the active tab and send "beastify.js" a message
-containing the URL to the chosen beast's image.
+Else if the click is on one of the clear buttons:
+  The text content of the node is the clear method we want.
+
+  Inject the "clear.js" content script in the active tab.
+
+  Then get the active tab and send "clear.js" a message
+  containing the clear method.
 */
 document.addEventListener("click", function(e) {
-  if (!e.target.classList.contains("beast")) {
-    return;
+  if (e.target.classList.contains("beast")) {
+    var chosenBeast = e.target.textContent;
+    var chosenBeastURL = beastNameToURL(chosenBeast);
+
+    chrome.tabs.executeScript(null, {
+      file: "/content_scripts/beastify.js"
+    });
+
+    chrome.tabs.query({active: true, currentWindow: true}, function(tabs) {
+      chrome.tabs.sendMessage(tabs[0].id, {beastURL: chosenBeastURL});
+    });
+
+    self.beast = true;
+
+    updateButtons();
   }
+  else if (e.target.classList.contains("clear")) {
+    var chosenMethod = e.target.textContent;
 
-  var chosenBeast = e.target.textContent;
-  var chosenBeastURL = beastNameToURL(chosenBeast);
+    chrome.tabs.executeScript(null, {
+      file: "/content_scripts/clear.js"
+    });
 
-  chrome.tabs.executeScript(null, {
-    file: "/content_scripts/beastify.js"
-  });
+    chrome.tabs.query({active: true, currentWindow: true}, function(tabs) {
+      chrome.tabs.sendMessage(tabs[0].id, {type: chosenMethod});
+    });
```

Comment:
If you use reloading to reset, you don't need to have a content script here. You can do it directly from the popup script using code like:

chrome.tabs.reload()

This will make things much easier!